### PR TITLE
Adding diagnostics tests to all modules

### PR DESCRIPTION
**Pull request by rahalan**
# Change

Adding diagnostics tests to all modules

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.DataFactory/factories/readme.md [hunk 1]
```diff
@@ -35,7 +37,7 @@
 | `publicNetworkAccess` | bool | `True` |  | Optional. Enable or disable public network access. |
 | `roleAssignments` | array | `[]` |  | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or it's fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11'. |
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
-| `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
+| `workspaceId` | string |  |  | Optional. Resource identifier of log analytics. |
```
Comment (MrMCake) [inline arm/Microsoft.DataFactory/factories/readme.md:40]: Should not be rolled back. Please update in bicep file and here

File: arm/Microsoft.Automation/automationAccounts/readme.md [hunk 2]
```diff
@@ -49,7 +49,43 @@ This module deploys an Azure Automation Account.
 | `tags` | object | `{object}` |  | Optional. Tags of the Automation Account resource. |
 | `userAssignedIdentities` | object | `{object}` |  | Optional. The ID(s) to assign to the resource. |
 | `variables` | _[variables](variables/readme.md)_ array | `[]` |  | Optional. List of variables to be created in the automation account. |
-| `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
+| `workspaceId` | string |  |  | Optional. Resource identifier of log analytics. |
```
Comment (MrMCake) [inline arm/Microsoft.Automation/automationAccounts/readme.md:52]: Should not be rolled back. Please update in bicep file and here

File: arm/Microsoft.EventHub/namespaces/readme.md [hunk 3]
```diff
@@ -43,9 +45,74 @@ This module deploys an EventHub namespace
 | `skuName` | string | `Standard` | `[Basic, Standard]` | Optional. event hub plan SKU name |
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
 | `vNetId` | string |  |  | Optional. Virtual Network ID to lock down the Event Hub. |
-| `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
+| `workspaceId` | string |  |  | Optional. Resource identifier of log analytics. |
```
Comment (MrMCake) [inline arm/Microsoft.EventHub/namespaces/readme.md:48]: Should not be rolled back. Please update in bicep file and here

File: arm/Microsoft.KeyVault/vaults/readme.md [hunk 4]
```diff
@@ -48,7 +47,7 @@ This module deploys a key vault and it's child resources.
 | `tags` | object | `{object}` |  | Optional. Resource tags. |
 | `vaultSku` | string | `premium` | `[premium, standard]` | Optional. Specifies the SKU for the vault |
 | `vNetId` | string |  |  | Optional. Virtual Network resource identifier, if networkAcls is passed, this value must be passed as well |
-| `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
+| `workspaceId` | string |  |  | Optional. Resource identifier of log analytics. |
```
Comment (MrMCake) [inline arm/Microsoft.KeyVault/vaults/readme.md:50]: Should not be rolled back. Please update in bicep file and here

File: arm/Microsoft.Automation/automationAccounts/readme.md [hunk 5]
```diff
@@ -49,7 +49,43 @@ This module deploys an Azure Automation Account.
...
+
+### Parameter Usage: `privateEndpoints`
```
[the remark this answers is not in the document]
Comment (rahalan) [reply arm/Microsoft.Automation/automationAccounts/readme.md:54]: ???

File: arm/Microsoft.DataFactory/factories/readme.md (hunk 1 above)
Comment (rahalan) [reply arm/Microsoft.DataFactory/factories/readme.md]: done

File: arm/Microsoft.Automation/automationAccounts/readme.md (hunk 2 above)
Comment (rahalan) [reply arm/Microsoft.Automation/automationAccounts/readme.md]: done

File: arm/Microsoft.EventHub/namespaces/readme.md (hunk 3 above)
Comment (rahalan) [reply arm/Microsoft.EventHub/namespaces/readme.md]: done

File: arm/Microsoft.KeyVault/vaults/readme.md (hunk 4 above)
Comment (rahalan) [reply arm/Microsoft.KeyVault/vaults/readme.md]: done

File: arm/Microsoft.Automation/automationAccounts/readme.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.Automation/automationAccounts/readme.md:54]: The section is twice in the file...

[the remark this answers is not in the document]
Comment (rahalan) [reply arm/Microsoft.Automation/automationAccounts/readme.md:54]: fixed

**Review by MrMCake — DISMISSED** [review]

**Review by MrMCake — APPROVED** [review]

**Pull request merged**